frontend/src/scenes/pipeline/Transformations.tsx

```diff
@@ -0,0 +1,372 @@
+                                                    to={plugins[pluginConfig.plugin].url}
+                                                    targetBlank={true}
+                                                    id={`app-${pluginConfig.id}-source-code`}
+                                                    fullWidth
+                                                >
+                                                    View app source code
+                                                </LemonButton>
+                                            )}
+                                            <LemonDivider />
+                                            <LemonButton
+                                                status="danger"
+                                                onClick={() => {
+                                                    deleteWithUndo({
+                                                        endpoint: `plugin_config`,
+                                                        object: { id: pluginConfig.id, name: pluginConfig.name },
+                                                        callback: loadPluginConfigs,
+                                                    })
+                                                }}
+                                                id={`app-reorder`}
+                                                disabledReason={
+                                                    canConfigurePlugins
+                                                        ? undefined
+                                                        : 'You do not have permission to delete apps.'
+                                                }
+                                                fullWidth
+                                            >
+                                                Delete app
+                                            </LemonButton>
+                                        </>
+                                    }
+                                />
+                            )
+                        },
+                    },
+                ]}
+                emptyState={
+                    <ProductIntroduction // TODO: maybe show this also when only GeoIP is enabled?
+                        productName="Pipeline transformations"
+                        thingName="transformation"
+                        productKey={ProductKey.PIPELINE_TRANSFORMATIONS}
+                        description="Pipeline transformations allow you to enrich your data with additional information, such as geolocation."
+                        docsURL="https://posthog.com/docs/cdp"
+                        actionElementOverride={<NewButton tab={PipelineTabs.Transformations} />}
+                        isEmpty={true}
+                    />
+                }
+            />
+        </>
+    )
+}
+
+function renderApp(plugin: PluginType): JSX.Element {
```

Comment:
There is a real functional difference, but I think it would fit React's declarative model & conventions better if this was a component with props.
```suggestion
type RenderAppProps = {
  plugin: PluginType
}

function RenderApp({ plugin }: RenderAppProps): JSX.Element {
```
used like

```tsx
<RenderApp plugin={plugin} />
```